src/Softeq.NetKit.Chat.Domain.Services/DomainServices/ChannelService.cs

```diff
@@ -289,33 +296,33 @@ public async Task<ChannelResponse> CloseChannelAsync(string saasUserId, Guid cha
 
         public async Task<IReadOnlyCollection<ChannelSummaryResponse>> GetAllowedChannelsAsync(string saasUserId)
         {
-            var creator = await UnitOfWork.MemberRepository.GetMemberBySaasUserIdAsync(saasUserId);
-            if (creator == null)
+            var currentUser = await UnitOfWork.MemberRepository.GetMemberBySaasUserIdAsync(saasUserId);
+            if (currentUser == null)
             {
                 throw new NetKitChatNotFoundException($"Unable to get allowed channels. Member {nameof(saasUserId)}:{saasUserId} is not found.");
             }
 
             var channelsResponse = new List<ChannelSummaryResponse>();
 
-            var channels = await UnitOfWork.ChannelRepository.GetAllowedChannelsWithMessagesAndCreatorAsync(creator.Id);
+            var channels = await UnitOfWork.ChannelRepository.GetAllowedChannelsWithMessagesAndCreatorAsync(currentUser.Id);
             foreach (var channel in channels)
             {
-                var channelMember = await UnitOfWork.ChannelMemberRepository.GetChannelMemberAsync(creator.Id, channel.Id);
+                var channelMember = await UnitOfWork.ChannelMemberRepository.GetChannelMemberAsync(currentUser.Id, channel.Id);
                 if (channel.Type == ChannelType.Direct)
                 {
                     //TODO: Extend existing query to make one single request to get all data for both direct and group channel
-                    var members = await UnitOfWork.ChannelMemberRepository.GetChannelMembersAsync(channel.Id);
-                    var directChannelMember = members.First(x => x.MemberId != creator.Id);
-                    var member = await UnitOfWork.MemberRepository.GetMemberByIdAsync(directChannelMember.MemberId);
+                    var members = await UnitOfWork.ChannelMemberRepository.GetChannelMembersWithMemberDetailsAsync(channel.Id);
+                    var directChannelMember = members.First(x => x.MemberId != currentUser.Id);
```

Comment:
Can we replace it with SingleOrDefault and throw an exception if it null?

Reply:
done